frontend/src/pages/OrderStatusPage.tsx

```diff
@@ -0,0 +1,38 @@
+import { useGetMyOrders } from '@/api/OrderApi';
+import OrderStatusDetails from '@/components/OrderStatusDetails';
+import OrderStatusHeader from '@/components/OrderStatusHeader';
+import { AspectRatio } from '@/components/ui/aspect-ratio';
+
+const OrderStatusPage = () => {
+  const { orders, isLoading } = useGetMyOrders();
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!orders || orders.length === 0) {
+    return <div>No orders found</div>;
+  }
+
+  return (
+    <div className="space-y-4">
+      {orders.map((order) => (
+        <div key={order._id} className="space-y-10 bg-gray-100 p-4 rounded-lg">
+          <OrderStatusHeader order={order} />
+          <div className="grid md:grid-cols-2 gap-4">
+            <OrderStatusDetails order={order} />
+            <AspectRatio ratio={16 / 9}>
+              <img
+                src={order.restaurant.imageFile}
+                alt={order.restaurant.restaurantName}
+                className="object-cover rounded-md h-full w-full"
+              />
+            </AspectRatio>
+          </div>
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting image rendering logic</b></div>
<div id="fix">

Consider extracting the image rendering logic into a separate component for better code organization and reusability. This could improve the overall structure of the `OrderStatusPage` component.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -22,10 +22,2 @@
            <div className="grid md:grid-cols-2 gap-4">
              <OrderStatusDetails order={order} />
 -            <AspectRatio ratio={16 / 9}>
 -              <img
 -                src={order.restaurant.imageFile}
 -                alt={order.restaurant.restaurantName}
 -                className="object-cover rounded-md h-full w-full"
 -              />
 -            </AspectRatio>
 +            <RestaurantImage restaurant={order.restaurant} />
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/49#issuecomment-2439903699>#0f11c6</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged